hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSPreemptionThread.java

```diff
@@ -204,6 +204,12 @@ private PreemptableContainers identifyContainersToPreemptOnNode(
     for (RMContainer container : containersToCheck) {
       FSAppAttempt app =
           scheduler.getSchedulerApp(container.getApplicationAttemptId());
+      if (app == null) {
+        // e.g. "INFO org.apache.hadoop.yarn.server.resourcemanager.scheduler.AbstractYarnScheduler: Container container_1536156801471_0071_01_000096 completed with event FINISHED, but corresponding RMContainer doesn't exist."
+        LOG.warn("app == null, giving up in identifyContainersToPreemptOnNode()");
+        return null;
```

Comment:
Should we just `continue` instead of returning `null` since we might still be able to find preemptable containers on this node?